moengage-segment-kotlin-destination/src/main/java/com/segment/analytics/kotlin/destinations/moengage/MoEngageDestination.kt

```diff
@@ -126,9 +126,24 @@ class MoEngageDestination(private val application: Application) : DestinationPlu
         try {
             super.identify(payload)
             Logger.print { "$tag identify(): will try to track $payload" }
+
             val traits = payload.traits
+            val internalTraits = removeTraitsWithNullValues(traits)
```

Comment:
this could be simplified as follows

```kotlin
traits.mapTransform(mapper).toContent().filter { entry ->
            entry.value != null
        } as MutableMap<String, Any>
``` 

With the above snippet, the method wouldn't be required.

Reply:
the variable name `internalTraits` isn't correct, can be something like `transformedTraits`

Reply:
Comments Addressed. Kindly have a look.